OpenRA.Mods.Common/Widgets/Logic/Settings/InputSettingsLogic.cs

```diff
@@ -152,17 +164,18 @@ Action ResetPanel(Widget panel)
 
 		public static void ShowMouseControlDropdown(DropDownButtonWidget dropdown, GameSettings gameSettings)
 		{
-			var options = new Dictionary<string, bool>()
+			var options = new Dictionary<string, MouseControlStyle>()
```

Comment:
This is duplicating `controlTypes`, and also keying on the button label is a bit odd. Not your fault, but could you please remove the `static` from this method and remove this in favour of `controlTypes`?

Reply:
Use this:
```csharp
public static void ShowMouseControlDropdown(DropDownButtonWidget dropdown, Dictionary<MouseControlStyle, string> controlTypes, GameSettings gameSettings)
{
	ScrollItemWidget SetupItem(MouseControlStyle o, ScrollItemWidget itemTemplate)
	{
		var item = ScrollItemWidget.Setup(itemTemplate,
			() => gameSettings.MouseControlStyle == o,
			() => gameSettings.MouseControlStyle = o);

		var label = controlTypes[o];
		item.Get<LabelWidget>("LABEL").GetText = () => label;
		return item;
	}

	dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 500, controlTypes.Keys, SetupItem);
}
```

It needs to stay static so it can be used by `IntroductionPromptLogic`.

Reply:
It either stays static or gets to use the controlTypes field, but not both

Reply:
ah, sorry, missed that